var/spack/repos/builtin/packages/spark/package.py

```diff
@@ -0,0 +1,36 @@
+from spack import *
+
+import distutils
+from distutils import dir_util
+
+
+class Spark(Package):
+    """
+    Apache Spark is a fast and general engine
+    for large-scale data processing.
+    """
+
+    homepage = "http://spark.apache.org"
+    url      = "http://spark.apache.org/downloads.html"
+
+    variant('hadoop', default=False,
+            description='Build with Hadoop')
+
+    depends_on('jdk')
```

Comment:
Because they're used in `setup_environment`, 'jdk' and 'hadoop' are both build and run time dependencies, since `setup_environment` is run at build time to generate modules/dotkits